pkg/api/steve/projectresources/handlers.go

```diff
@@ -0,0 +1,853 @@
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/wrangler/pkg/apply"
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+	authzv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientauthzv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	apiregv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+)
+
+const (
+	// Group is the APIGroup for this APIService.
+	Group = "resources.project.cattle.io"
+	// ParentLabel is the label assigned to namespaces that represent Projects.
+	ParentLabel = "cattle.io/parent"
+	// RoleName is the name of the ClusterRole and RoleBindings used to grant access to this API.
+	RoleName = "cattle-project-resources"
+	// AuthzAnnotation is the annotation designated for RBAC resources.
+	AuthzAnnotation = "resources.project.cattle.io/authz"
+	// NamespaceAnnotation is the annotation designated for namespaces.
+	NamespaceAnnotation = "resources.project.cattle.io/namespace"
+
+	apiServiceName             = version + "." + Group
+	groupVersion               = Group + "/" + version
+	version                    = "v1alpha1"
+	workers                    = int64(3)
+	queryKey                   = "fieldSelector"
+	projectsOrNamespacesKey    = "projectsornamespaces"
+	projectsOrNamespacesVar    = "projectsOrNamespaces"
+	cattleClusterAgentService  = "cattle-cluster-agent"
+	rancherService             = "rancher"
+	priority                   = 1 // low priority
+	internalCASecret           = "tls-rancher-internal-ca"
+	unscopedNamespace          = "cattle-unscoped"
+	notOp                      = "!"
+	extensionConfigMap         = "extension-apiserver-authentication"
+	kubeSystemNamespace        = "kube-system"
+	clientCAKey                = "requestheader-client-ca-file"
+	serverAllowedCNKey         = "requestheader-allowed-names"
+	projectNamespaceAnnotation = "management.cattle.io/system-namespace"
+)
+
+var (
+	paramScheme             = runtime.NewScheme()
+	paramCodec              = runtime.NewParameterCodec(paramScheme)
+	queryOpReg              = regexp.MustCompile("[=!]?=")
+	queryValue              = fmt.Sprintf("%s{op:[=!]?}={%s}", projectsOrNamespacesKey, projectsOrNamespacesVar)
+	orphanNamespaceSelector labels.Selector
+)
+
+func init() {
+	metav1.AddToGroupVersion(paramScheme, metav1.SchemeGroupVersion)
+
+	var err error
+	selectorString := "!" + nslabels.ProjectIDFieldLabel
+	orphanNamespaceSelector, err = labels.Parse(selectorString)
+	if err != nil {
+		panic(fmt.Sprintf("programmer error: invalid selector %s", selectorString))
+	}
```

Comment:
Nit: This might be more readable as:

```golang
requirement, err := labels.NewRequirement(nslabels.ProjectIDFieldLabel, selection.DoesNotExist, nil)
if err != nil {
	panic(fmt.Sprintf("programmer error: invalid selector %s", selectorString))
}
orphanNamespaceSelector = labels.NewSelector().Add(*requirement)
```